core/dir/empty_spec.rb

```diff
@@ -0,0 +1,37 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../fixtures/common', __FILE__)
+
+describe "Dir.empty?" do
+  ruby_version_is "2.4" do
+    before :all do
+      DirSpecs.create_mock_dirs
+      @empty_dir = DirSpecs.mock_dir "empty_dir"
+      mkdir_p @empty_dir
+    end
+
+    after :all do
+      DirSpecs.delete_mock_dirs
+      rm_r @empty_dir
+    end
+
+    it "returns true for empty directories" do
+      result = Dir.empty? @empty_dir
+      result.should be_true
+    end
+
+    it "returns false for non-empty directories" do
+      result = Dir.empty? DirSpecs.mock_dir
+      result.should be_false
+    end
+
+    it "returns false for a non-directory" do
+      first_file = DirSpecs.mock_dir_files.first
+      result = Dir.empty? "#{DirSpecs.mock_dir}/#{first_file}"
+      result.should be_false
```

Comment:
I like the idea from @esparta to use `__FILE__` and `__dir__` here and in the `it` above, which require no DirSpecs mock dirs.
Could you switch to using those? Then the `before :all` only needs to create the `@empty_dir`.